keylime/src/structures/capabilities_negotiation.rs

```diff
@@ -49,19 +52,29 @@ pub struct ImaEntries {
 }
 
 #[derive(Serialize, Deserialize, Debug)]
-pub struct AgentCapabilities {
-    spec_version: String,
+pub struct Capabilities {
+    component_version: String,
     hash_algorithms: Vec<String>,
-    signing_schemes: Vec<String>,
-    attestation_keys: Vec<AttestationKey>,
+    signature_schemes: Vec<String>,
+    available_subjects: ShaValues,
+    certification_keys: Vec<CertificationKey>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+// Do not serialize the struct name, only the fields
+pub struct ShaValues {
+    sha1: Vec<u8>,
+    sha256: Vec<u8>,
 }
 
 #[derive(Serialize, Deserialize, Debug, Clone)]
-pub struct AttestationKey {
+pub struct CertificationKey {
     key_class: String,
```

Comment:
In my opinion, `key_class` is redundant with `key_algorithm` as the type of the key (symmetric of asymmetric) is implicit in the `key_algorithm`